### PR TITLE
Feature/yash jenny/button component

### LINKED ISSUE #3: Button Component
## What do we need?
Create a button component that can be reused all across the app. This component should be generic so that without any changes to the component it can be shared by various pages

- We need 3 types of buttons (filled active, outlined active, filled disabled). 
- Button should be resizable
image

## Example uses

Filled button in active state
image

Filled button in disabled state
image

Filled and Outlined Button in active state (resizable)
image

## Resources
You are encouraged to look for more resources if needed, but here some I have - 

https://www.kodeco.com/34851726-swiftui-button-tutorial-customization#toc-anchor-008
Shows ButtonStyle protocol around 10:59 - https://youtu.be/Kt4vpQ9ms5E?t=659
Look into Apple's ButtonStyle protocol - https://developer.apple.com/documentation/swiftui/buttonstyle

https://betterprogramming.pub/create-custom-button-styles-in-swiftui-f3778f695266

### PR DESCRIPTION
**Pull request by YashRavipati1**
### Tracking Info

Resolves #3

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

Created the resizable Button Components for the FilledInButton and OutlinedButton. Also made these buttons have a state to be disabled.

### Testing

How did you confirm your changes worked? 

We added previews in Xcode to ensure that the Buttons looked right and were pressable/not pressable in the
enabled and disabled states respectively.

### Confirmation of Change 

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

 
image

image

image

image

image

image

### CODE REVIEW

**Comment by AmanKAggarwal** [comment]
Can we add screenshots of the outlines button too? @jennymar @YashRavipati1

File: ALUM/ALUM/Components/FilledButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/FilledButton.swift:21]: Giving this more thought, hardcoding widths for the buttons don't seem ideal. And given our use cases it will be best if we deal with their widths while using the component. 

Instead of hardcoding in the widths, let's use `.frame` to set the minWidth to 0 and maxWidth to .infinity. This will allow our button width to be resizable based on its parent.  If we do this, you can remove the width state of the button. 

For button heights, we don't have any other option but to hardcode it so feel free to do that. See if you can using minHeight and maxHeight can make our button more flexible but I'm not too worried about fixing the height.

Idea is that if you put two buttons with our styles inside an HStack, the buttons should resize well to co-exist in the same space. Lmk if you have any questions!

File: ALUM/ALUM/Components/FilledButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/FilledButton.swift:24]: Instead of .system(size: 17, weight: .medium) can we use .body to support Apple's dynamic type. 

Apple has this feature called dynamic type which lets users set their preferred font sizes (out of few categories) which are then used across most apps. 

Check this out for understanding dynamic types (especially the table) - 
https://developer.apple.com/design/human-interface-guidelines/foundations/typography/#:~:text=xxxLarge-,Large%20(Default),13,-Point%20size%20based

File: ALUM/ALUM/Components/FilledButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/FilledButton.swift:31]: Same note about making width flexible

File: ALUM/ALUM/Components/FilledButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/FilledButton.swift:53]: Once you are done, create a preview with two buttons inside an HSTack. Could be something like - 
```{swift}
HStack(spacing: 10) {
            FilledButton()
                .frame(minWidth: 50, maxWidth: 300, alignment: .leading)
            FilledButton()
                .frame(minWidth: 50, maxWidth: 150, alignment: .leading)
        }
        .padding(10)
```
This example should basically render two buttons co-existing proportionately in the same horizontal area.

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/OutlinedButton.swift:14]: Apply the changes suggested about width in FilledButton

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/OutlinedButton.swift:11]: This state would no longer be needed

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/OutlinedButton.swift:2]: Let's rename this file to OutlinedButtonStyle

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/OutlinedButton.swift:24]: Same suggestion as FilledButton

**Review by AmanKAggarwal — CHANGES_REQUESTED** [review]
Awesome PR. Just few minor style updates to make the buttons more flexible but other than that this PR should be good to merge. Ping me once you fix these and I can approve it!

**Review by sidhantrohatgi — APPROVED** [review]

File: ALUM/ALUM/Components/FilledInButtonStyle.swift
Comment (AmanKAggarwal) [inline ALUM/ALUM/Components/FilledInButtonStyle.swift:17]: nit I think button height should go till infinity because we want all our buttons to be of consistent size. You can just have this as fixed size if that's easier

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [reply ALUM/ALUM/Components/OutlinedButton.swift:2]: Can we do this as well? With the change of button height from Infinity to fixed

Comment (AmanKAggarwal) [reply ALUM/ALUM/Components/OutlinedButton.swift:2]: Just realised, you added a OutlinedButtonStyle. You missed deleting this OutlinedButton.swift though so let's remove that.

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [reply ALUM/ALUM/Components/OutlinedButton.swift:11]: Fixed in OutlinedButtonStyle so this file can be deleted

File: ALUM/ALUM/Components/OutlinedButton.swift
Comment (AmanKAggarwal) [reply ALUM/ALUM/Components/OutlinedButton.swift:14]: Done in OutlinedButtonStyle

**Review by AmanKAggarwal — COMMENTED** [review]
Just need the OutlinedButton to be removed and the button heights can be made fix so that our button heights are consistent throughout in the app.

It's good to merge after that! Just ping on slack when you update though. GitHub doesn't send me notifications lol :(

**Review by AmanKAggarwal — APPROVED** [review]
Looks good to merge!

**Pull request merged**
